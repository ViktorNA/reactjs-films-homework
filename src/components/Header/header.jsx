import React from 'react';
import styles from './header.scss';

const placeholder = require('../../assets/json/505954.json').title;

const Header = () => (
  <header className={styles.header}>
    <div className={styles.logo}>FILMS</div>
    <div className={styles.searchBox}>
      <input className={styles.searchInput} placeholder={placeholder} />
    </div>
  </header>
);

export default Header;