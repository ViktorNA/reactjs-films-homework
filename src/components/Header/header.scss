.header {
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 100%;
  height: 100px;
  background-image: linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0));
  z-index: 999;
  position: fixed;
}

.logo {
  margin-left: 70px;
  margin-top: 15px;
  color: white;
  font-size: 25px;
  text-align: left;
}

.searchBox {
  text-align: left;
  align-items: center;
  border: 2px solid white;
  border-radius: 6px;
  width: 270px;
  height: 30px;
  justify-self: flex-end;
  background-image: url(../../assets/images/Search.svg);
  background-repeat: no-repeat;
  background-position: 240px;
  background-size: 20px 20px;
  margin-top: 15px;
  margin-right: 70px;
}

.searchInput {
  color: white;
  border: none;
  height: inherit;
  width: 220px;
  font-size: 14px;
  padding-left: 10px;
  outline: 0;
  background-color: rgba(0, 0, 0, 0);
}

.searchInput::placeholder {
  color: white;
}
