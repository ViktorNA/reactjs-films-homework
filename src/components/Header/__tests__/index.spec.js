import TestRenderer from 'react-test-renderer';
import React from 'react';
import Header from '..';

describe('Snapshot test', () => {
  it('works correctly', () => {
    const testRender = TestRenderer.create(<Header />);
    expect(testRender.toJSON()).toMatchSnapshot();
  });
});
