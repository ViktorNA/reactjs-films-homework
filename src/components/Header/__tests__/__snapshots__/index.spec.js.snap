// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test works correctly 1`] = `
<header
  className="header"
>
  <div
    className="logo"
  >
    FILMS
  </div>
  <div
    className="searchBox"
  >
    <input
      className="searchInput"
      placeholder="T-34"
    />
  </div>
</header>
`;
